src/shared/lib/JsonPathUtils.ts

```diff
@@ -5,21 +5,28 @@ export function hasJsonPathPlaceholders(message: String) {
 
 export function replaceJsonPathPlaceholders(
     message: String,
-    studyMetadata: any
+    studyMetadata: any,
+    studyId: String
 ) {
     let placeholders = message.match(/[{][$][^}]*[}]/g);
     let placeholdersReplaced: boolean = true;
     if (placeholders !== null) {
         let jp = require('jsonpath');
         placeholders.forEach(placeholder => {
-            let placeholderReplaceValue = jp.query(
-                studyMetadata,
-                placeholder.replace(/["'{}]/g, '')
-            );
-            if (placeholderReplaceValue.length === 0)
+            let placeholderReplaceValue = '';
+            if (placeholder === '{$.studyId}') {
+                placeholderReplaceValue = studyId.toString();
```

Comment:
StudyId is a string already. Why use `toString()`?

Reply:
Fixed